services/src/main/java/org/exoplatform/leadcapture/listeners/NewResponseListener.java

```diff
@@ -12,60 +15,96 @@
-import org.exoplatform.leadcapture.Utils;
+import org.exoplatform.leadcapture.services.ResourcesManagementService;
 import org.exoplatform.services.listener.Event;
 import org.exoplatform.services.listener.Listener;
 import org.exoplatform.services.log.ExoLogger;
 import org.exoplatform.services.log.Log;
 
 public class NewResponseListener extends Listener<LeadEntity, ResponseEntity> {
 
-  private static final Log        LOG = ExoLogger.getLogger(NewResponseListener.class);
+  private static final Log               LOG = ExoLogger.getLogger(NewResponseListener.class);
 
-  private LCMailService           lcMailService;
+  private LCMailService                  lcMailService;
 
   private MailTemplatesManagementService mailTemplatesManagementService;
 
-  private FieldDAO fieldDAO;
+  private ResourcesManagementService     resourcesManagementService;
 
-  public NewResponseListener(LCMailService lcMailService, MailTemplatesManagementService mailTemplatesManagementService, FieldDAO fieldDAO) {
+  private FieldDAO                       fieldDAO;
+
+  private LeadCaptureSettingsService     leadCaptureSettingsService;
+
+  public NewResponseListener(LCMailService lcMailService,
+                             MailTemplatesManagementService mailTemplatesManagementService,
+                             ResourcesManagementService resourcesManagementService,
+                             LeadCaptureSettingsService leadCaptureSettingsService,
+                             FieldDAO fieldDAO) {
     this.lcMailService = lcMailService;
     this.mailTemplatesManagementService = mailTemplatesManagementService;
+    this.resourcesManagementService = resourcesManagementService;
+    this.leadCaptureSettingsService = leadCaptureSettingsService;
     this.fieldDAO = fieldDAO;
   }
 
   @Override
   public void onEvent(Event<LeadEntity, ResponseEntity> event) throws Exception {
     LeadEntity lead = event.getSource();
+    String field = null;
     ResponseEntity responseEntity = event.getData();
     List<MailTemplateEntity> templates = mailTemplatesManagementService.getTemplatesbyEvent("newResponse");
     for (MailTemplateEntity template : templates) {
       MailContentDTO content = null;
       MailTemplateDTO mailTemplateDTO = mailTemplatesManagementService.toMailTemplateDTO(template);
       if (mailTemplateDTO.getContents().size() > 0) {
         content = Utils.getContentForMail(mailTemplateDTO, lead);
-        if (content != null && shouldBeSend(template, responseEntity)) {
-          lcMailService.sendMail(content.getContent(), content.getSubject(), lead);
-          LOG.info("service=lead-capture operation=send_mail_to_lead parameters=\"lead_id:{},mail_template_id:{},mail_template_name:{},reason: NewLead\"", lead.getId(), mailTemplateDTO.getId(), mailTemplateDTO.getName());
+        if (content != null) {
+          field = getField(template, responseEntity);
+          if (field != null) {
+            if (!field.equals("")) {
```

Comment:
Could be combined in one test for null and return null instead of empty string in getField()